addons/stock_account/models/stock_quant.py

```diff
@@ -64,7 +64,7 @@ def read_group(self, domain, fields, groupby, offset=0, limit=None, orderby=Fals
     def _apply_inventory(self):
         acc_inventories = self.filtered(lambda quant: quant.accounting_date)
         for inventory in acc_inventories:
-            super(StockQuant, self.with_context(force_period_date=inventory.accounting_date))._apply_inventory()
+            super(StockQuant, inventory.with_context(force_period_date=inventory.accounting_date))._apply_inventory()
```

Comment:
What about grouping them by date and _apply_inventory on a RecordSet? You'll speel up the process even more.